pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -114,6 +116,12 @@ func (s *ServiceEntryStore) workloadEntryHandler(old, curr config.Config, event
 		namespace: curr.Namespace,
 	}
 
+	// If an entry is unhealthy, we will mark this as a delete instead
+	// This ensures we do not track unhealthy endpoints
+	if !isHealthy(curr) {
```

Comment:
Not sure if there are any, but in the case where we actually care about an Add vs. Update, should we convert Update -> Add if going from Unhealthy to Healthy?